snapcraft/internal/elf.py

```diff
@@ -86,52 +84,94 @@ def _crawl_for_path(*, soname: str, root_path: str,
             for file_name in files:
                 if file_name == soname:
                     file_path = os.path.join(root, file_name)
-                    if _is_dynamically_linked_elf(_get_magic(file_path)):
+                    if ElfFile.is_elf(file_path):
                         return file_path
     return None
 
 
 class ElfFile:
     """ElfFile represents and elf file on a path and its attributes."""
 
-    def __init__(self, *, path: str, magic: str) -> None:
+    @classmethod
+    def is_elf(cls, path: str) -> bool:
+        with open(path, 'rb') as bin_file:
+            return bin_file.read(4) == b'\x7fELF'
+
+    def __init__(self, *, path: str) -> None:
         """Initialize an ElfFile instance.
 
         :param str path: path to an elf_file within a snapcraft project.
-        :param str magic: the magic string for path.
         """
         self.path = path
-        self.is_executable = 'interpreter' in magic
         self.dependencies = set()  # type: Set[Library]
-        self.symbols = self._get_symbols(path)  # type: List[Symbol]
+        self._type, self.symbols = self._extract_readelf(path)
+
+    def is_executable(self):
+        return self._type == 'EXEC'
+
+    def is_shared_object(self):
+        return self._type == 'DYN'
 
-    def _get_symbols(self, path) -> List[Symbol]:
+    def _extract_readelf(self, path) -> Tuple[str, List[Symbol]]:
+        file_type = str()
         symbols = list()  # type: List[Symbol]
-        symbols_section = subprocess.check_output([
-            'readelf', '--wide', '--dyn-syms', path]).decode()
+        output = subprocess.check_output([
+            'readelf', '--wide', '--file-header', '--dyn-syms', path])
+        readelf_lines = output.decode().split('\n')
         # Regex inspired by the regex in lintian to match entries similar to
         # number '1' from the following sample output
```

Comment:
This comment may be out of place, now. Or perhaps needs to be rewritten. The '1' to which you're referring is unclear.